mlir/lib/Quantum/Transforms/LoopBoundaryOptimizationPatterns.cpp

```diff
@@ -139,6 +139,17 @@ bool hasQuantumCustomPredecessor(CustomOp &op)
     return false;
 }
 
+// Return true if all operands of topOp are immediately after bottomOp in the IR
+bool isSourceAndSink(CustomOp topOp, CustomOp bottomOp)
```

Comment:
Personally I don't find `isSourceAndSink` very clear if conceptually what we are trying to find out is if one op is the immediate successor of the other, but since this was already a code review suggestion we can leave it as is.
I'd adjust the comment though:
```suggestion
// Return true if bottomOp is the immediate successor of topOp.
// In this case, we don't want to perform loop boundary commutation,
// as it would prevent a fix-point from being reached by the rewrite driver.
bool isSourceAndSink(CustomOp topOp, CustomOp bottomOp)
```

Reply:
Thanks, @dime10! Comment looks clearer. I’d keep the name for now but open to change it later if needed.